koku/masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -57,16 +57,16 @@ def get_initial_dataframe_with_date(
     invoice_bill = "bill/InvoiceId"
     time_interval = "identity/TimeInterval"
     optional_cols = ["resourcetags", "costcategory"]
-    base_cols = utils.INGRESS_REQUIRED_COLUMNS
     try:
         data_frame = pd.read_csv(local_file, usecols=[invoice_bill], nrows=1)
     except ValueError:
         invoice_bill = "bill_invoice_id"
         time_interval = "identity_time_interval"
         optional_cols = ["resource_tags", "cost_category"]
-        base_cols = utils.INGRESS_ALT_COLUMNS
-    use_cols = com_utils.fetch_optional_columns(local_file, base_cols, optional_cols, tracing_id, context)
-    data_frame = pd.read_csv(local_file, usecols=use_cols)
+    # drop tag/categry columns that only contain null values.
+    drop_cols = com_utils.list_null_optional_columns(local_file, optional_cols, tracing_id, context)
+    headers = [*pd.read_csv(local_file, nrows=1)]
+    data_frame = pd.read_csv(local_file, usecols=[c for c in headers if c not in drop_cols])
```

Comment:
I feel like we're opening this CSV so many times using pandas. We open it twice right here after opening the entire CSV and doing some processing on it in `list_null_optional_columns`. 

Could we not just open it once and drop the null columns instead of opening it and doing processing on it before we then open a subset of columns?